src/transcript_tool.py

```diff
@@ -56,39 +88,153 @@ def main(in_path: str, out_path: str, model_path: str) -> None:
 
     # fetch model
     model = SpeechRecognitionModel(model_path)
-    print(f"\nModel: {model_path} fetched\n")
+    logging.info("Model loaded from: %s", model_path)
+
+    # create interim folders for processing slices & segments
+    if not os.path.isdir("../data/interim/slices"):
+        os.makedirs(os.path.abspath("../data/interim/slices"))
+    if not os.path.isdir("../data/interim/segments"):
+        os.makedirs(os.path.abspath("../data/interim/segments"))
```

Comment:
TODO: add test; package install can run into troubles with permissions
Would be nicer to check permissions, too and explicitly raise exceptions (e.g. if permission is denied and etc).

Reply:
i remember that we talked about this, but im not sure how i can create such a scenario. how do i force a permission error if the ../data folder is created by and within the package itself?

Reply:
moved interim folders under the output folder, also slicing and segmenting gets their output paths as an argument, so they are easier to test as well. currently writing the tests.